app/src/org/commcare/network/HttpUtils.java

```diff
@@ -68,4 +75,17 @@ private static String getCredential(String username, String password) {
             return Credentials.basic(username, buildAppPassword(password));
         }
     }
+
+    public static String parseUserVisibleError(Response<ResponseBody> response) {
+        String message;
+        try {
+            JSONObject errorKeyAndDefault = new JSONObject(response.errorBody().string());
+            message = Localization.getWithDefault(
+                    errorKeyAndDefault.getString("error"),
+                    errorKeyAndDefault.getString("default_response"));
+        } catch (JSONException | IOException e) {
+            message = "Unknown issue";
```

Comment:
I wonder whether it would be a net positive in this case to just publish the errorBody itself as the error? A 406 is specific enough that I feel like we'll be happy in the event of an issue to be able to dig into the error itself.

Reply:
Nice point. Will update.